viewer/src/utilities/networking/types.ts

```diff
@@ -0,0 +1,27 @@
+/*!
+ * Copyright 2020 Cognite AS
+ */
+
+import { File3dFormat } from '../File3dFormat';
+
+import { IdEither } from '@cognite/sdk';
+
+export interface BlobOutputMetadata {
+  blobId: number;
+  format: File3dFormat | string;
+  version: number;
+}
+
+export interface ModelUrlProvider<TModelIdentifier> {
+  getModelUrl(params: TModelIdentifier): Promise<string>;
+}
+
+export interface ByUrlModelIdentifier {
+  discriminator: 'external';
```

Comment:
Why do we have discriminator here? And why does this have to be an entire url. I liked the restriction the local host imposed by locking the url to the origin of the url. It meant people only fed the path and not the entire url.

Reply:
It's here because it's a type? Before it was in the same folder, just another file.

As for URL in the name it has an URL-property so if the property is used as a relative path that property must be renamed.

I don't agree that locking to localhost has any value though. A use case would be to have some automated test with datasets stored externally to avoid a ton of blobs in git.